druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -241,6 +241,14 @@ public static DruidQuery extendQuery(DruidQuery query,
         query.druidTable, intervals, query.rels, query.getOperatorConversionMap());
   }
 
+  /** Check if it is needed to use UTC for DATE and TIMESTAMP types. **/
+  private static boolean needUTCTimeExtract(RexNode rexNode) {
+    return rexNode.getType().getSqlTypeName() == SqlTypeName.DATE
+        || rexNode.getType().getSqlTypeName() == SqlTypeName.TIMESTAMP
+        || rexNode.getType().getSqlTypeName()
+        == SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE;
```

Comment:
using switch case might be better. but it's up to you.

Reply:
can you rename the function to `needUtcTimeExtract`. house style is to camel-case acronyms such as 'UTC' and 'SQL' into 'Utc' and 'Sql'

Reply:
renamed, thx for the review, Julian